index.rb

```diff
@@ -1,21 +1,23 @@
+# frozen_string_literal: true
+
 require 'date'
 require 'json'
 
 require_relative 'library'
 
+l = Library.new
 
-# define_method(:convert) do |object| 
-# 	my_library[object] = object.to_json
-
-
-# File.open("library.json", "a") { |file| file.write(JSON.pretty_generate(my_library))}
+l.add_instance(a = Author.new('darina'))
+l.add_instance(b = Book.new('title', a))
```

Comment:
let's divide into 2 parts - first is creating of entities, second - adding entities to library